### DISCUSSION TITLE
Nordvpn over raspberry pi 5 with RaspAP... Nordvpn is connected but traffic is not routed over VPN

### DISCUSSION THREAD

**Discussion by Shilyjunior** [RaspAP/raspap-webgui#1562] [Q&A] [answered]
I am new to RaspAP and Raspberry pi projects in general. I'm using RaspAP with two WiFi networks: wlan1 for the AP and wlan0 as the client connecting to a network with access to the Internet. This works great but traffic is not routed through VPN when the VPN is connected. See attached VPN settings.

<img width="301" alt="Screenshot 2024-03-31 at 10 25 37" src="https://github.com/RaspAP/raspap-webgui/assets/165553275/7f245fa2-ebcc-4535-b3ba-3b28173bdb39">

**Comment by dlew56** [comment]
I have the same issue (with eth0 and wlan0).

**Reply by VecchieAbitudini** [reply]
> I have the same issue (with eth0 and wlan0).

Me too, any detailed guide how to resolve ?

**Reply by frankozland** [reply]
with nord installed and active can you please post the output of "ip a"
zilch out the ip addresses - im curious to see what nord is presenting as the tunnel
thanks

**Comment by billz** [comment]
Per the VPN providers docs on [control scope](https://docs.raspap.com/providers/#control-scope).

> RaspAP may be used to administer basic functions including connect, disconnect, status, account information and country (or city) selection for the remote VPN server.

For everything else, you'll have better luck with your provider's documentation / forums.

**Reply by dlew56** [reply, marked as answer]
I ended up uninstalling RaspAP and re-installing it with only the OpenVPN provider (instead of also NordVPN) and connecting via a NordVPN service credentials and it appears to now be capturing the client traffic.

**Reply by billz** [reply]
I've updated the provider doc page to reflect this, thanks.

**Reply by VecchieAbitudini** [reply]
> I've updated the provider doc page to reflect this, thanks.

Could you please post the link where I can find it ?

**Reply by billz** [reply]
https://docs.raspap.com/providers/

**Comment by frankozland** [comment]
hey Bill - for the other providers cli's you might need to add a parm to the cli json file.
Nord tunnel is "nordlynx".
Im sure each provider has their own quirk tunnel.

in config/vpn-providers.json, would it make sense to add a field for the tunnel name?

And then on up set the masquerade to match that cli field?
ie
```        {
            "id": 3,
            "name": "NordVPN",
            "bin_path": "/usr/bin/nordvpn",
            "tun_device":"nordlynx",
            "install_page": "https://nordvpn.com/download/linux/",
            "account_page": "https://my.nordaccount.com/dashboard/",
            "cmd_overrides": {
                "log": "status"
            },
```

so on masquerade step, just pass $tun_device?
```
iptables -t nat -A POSTROUTING -o $tun_device -j MASQUERADE
```

This may also be useful for monitoring - if $tun_device disappears from "ip a" the tunnel is dead.
